cmd/generator/generator.go

```diff
@@ -0,0 +1,339 @@
+		return o.value
+	}
+	return defaultValue()
+}
+
+func (o {{.Name}}) EncodeMsgpack(encoder *msgpack.Encoder) error {
+	if o.exists {
+		return {{ .EncoderFunc }}(encoder, o.value)
+	}
+	return encoder.EncodeNil()
+}
+
+func (o *{{.Name}}) DecodeMsgpack(decoder *msgpack.Decoder) error {
+	code, err := decoder.PeekCode()
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case code == msgpcode.Nil:
+		o.exists = false
+		return decoder.Skip()
+	case {{ .CheckerFunc }}(code):
+		o.exists = true
+		o.value, err = {{ .DecodeFunc }}(decoder)
+		return err
+	default:
+		return newFailedToDecodeError("{{.Name}}", code)
+	}
+}
+`
+
+func printFile(prefix string, data []byte) {
+	for _, line := range bytes.Split(data, []byte("\n")) {
+		fmt.Println(prefix + string(line))
+	}
+}
+
+func generateAndWrite() error {
+	tmpl, err := template.New("internal").Parse(tplText)
+	if err != nil {
+		return fmt.Errorf("failed to parse template: %w", err)
+	}
+
+	outputData := make(map[string][]byte, len(DefaultTypes))
+
+	var data bytes.Buffer
+
+	// 1. Generate code for each type.
+	for _, generatedType := range DefaultTypes {
+		tmplData := structToMap(generatedType)
+		tmplData["packageName"] = "option" // Package name is option, since generator is used for option only right now.
+		tmplData["imports"] = []string{}   // No additional imports are needed right now.
+
+		err := tmpl.Execute(&data, tmplData)
+		if err != nil {
+			return fmt.Errorf("failed to execute template: %w", err)
+		}
+
+		outputData[generatedType.Name] = slices.Clone(data.Bytes())
+		data.Reset()
+	}
+
+	// 2. Format code using gofmt.
+	for name, origData := range outputData {
+		data, err := format.Source(origData)
+		if err != nil {
+			if verbose {
+				printFile("> ", origData)
+			}
+			return fmt.Errorf("failed to format code: %w", err)
+		}
+
+		outputData[name] = data
+	}
+
+	// 3. Write resulting code to files.
+	for name, data := range outputData {
+		err = os.WriteFile(filepath.Join(outputDirectory, name+".go"), data, defaultGoPermissions)
+		if err != nil {
+			return fmt.Errorf("failed to write file: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func main() {
+	flag.StringVar(&outputDirectory, "output", ".", "output directory")
+	flag.BoolVar(&verbose, "verbose", false, "print verbose output")
+	flag.Parse()
+
+	// Get absolute path for output directory.
+	absOutputDirectory, err := filepath.Abs(outputDirectory)
+	if err != nil {
+		fmt.Printf("failed to get absolute path for output directory (%s): %s\n", outputDirectory, err.Error())
+		os.Exit(1)
+	}
+
+	// Check if output directory exists and is directory.
+	switch fInfo, err := os.Stat(absOutputDirectory); {
+	case err != nil:
+		fmt.Println("failed to stat output directory: ", err.Error())
+		os.Exit(1)
+	case !fInfo.IsDir():
+		fmt.Printf("output directory '%s' is not a directory\n", absOutputDirectory)
+		os.Exit(1)
+	}
+
+	err = generateAndWrite()
+	if err != nil {
+		fmt.Println("failed to generate or write code: ", err.Error())
```

Comment:
It should work fine.
```suggestion
		fmt.Println("failed to generate or write code: ", err)
```